src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -908,5 +915,296 @@ .maxstack  4
+            yield return new object[] { "MyColor_Int64", "MyColor_Int64.Red, MyColor_Int64.Blue", 2, "ldind.i8", "<PrivateImplementationDetails>.__StaticArrayInitTypeSize=16", "<PrivateImplementationDetails>.F7548C023E431138B11357593F5CCEB9DD35EB0B0A2041F0B1560212EEB6F13E8" };
+            yield return new object[] { "MyColor_UInt64", "MyColor_UInt64.Red, MyColor_UInt64.Blue", 2, "ldind.i8", "<PrivateImplementationDetails>.__StaticArrayInitTypeSize=16", "<PrivateImplementationDetails>.F7548C023E431138B11357593F5CCEB9DD35EB0B0A2041F0B1560212EEB6F13E8" };
+        }
+
+        [Theory]
+        [MemberData(nameof(CachedArray_NonSize1Type_MemberData))]
+        public void CachedArrayIsUsedForSpanOfNonSize1Types_NoCreateSpan(string type, string args, int length, string ldind, string fieldType, string fieldName)
+        {
+            string csharp = @$"
+public enum MyColor_Byte : byte {{ Red, Orange, Yellow, Green, Blue }}
+public enum MyColor_SByte : sbyte {{ Red, Orange, Yellow, Green, Blue }}
+public enum MyColor_UInt16 : ushort {{ Red, Orange, Yellow, Green, Blue }}
+public enum MyColor_Int16 : short {{ Red, Orange, Yellow, Green, Blue }}
+public enum MyColor_UInt32 : uint {{ Red, Orange, Yellow, Green, Blue }}
+public enum MyColor_Int32 : int {{ Red, Orange, Yellow, Green, Blue }}
+public enum MyColor_UInt64 : ulong {{ Red, Orange, Yellow, Green, Blue }}
+public enum MyColor_Int64 : long {{ Red, Orange, Yellow, Green, Blue }}
+
+public class Test
+{{
+    public static {type} M()
+    {{
+        System.ReadOnlySpan<{type}> s = new {type}[] {{ {args} }};
+        return s[0];
+    }}
+}}";
+
+            var compilation = CreateCompilationWithMscorlibAndSpan(csharp, TestOptions.ReleaseDll);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("Test.M", @$"{{
+  // Code size       48 (0x30)
+  .maxstack  3
+  .locals init (System.ReadOnlySpan<{type}> V_0) //s
+  IL_0000:  ldsfld     ""{type}[] {fieldName}_Array""
+  IL_0005:  dup
+  IL_0006:  brtrue.s   IL_0020
+  IL_0008:  pop
+  IL_0009:  ldc.i4.{length}
+  IL_000a:  newarr     ""{type}""
+  IL_000f:  dup
+  IL_0010:  ldtoken    ""{fieldType} {fieldName}""
+  IL_0015:  call       ""void System.Runtime.CompilerServices.RuntimeHelpers.InitializeArray(System.Array, System.RuntimeFieldHandle)""
+  IL_001a:  dup
+  IL_001b:  stsfld     ""{type}[] {fieldName}_Array""
+  IL_0020:  newobj     ""System.ReadOnlySpan<{type}>..ctor({type}[])""
+  IL_0025:  stloc.0
+  IL_0026:  ldloca.s   V_0
+  IL_0028:  ldc.i4.0
+  IL_0029:  call       ""ref readonly {type} System.ReadOnlySpan<{type}>.this[int].get""
+  IL_002e:  {ldind}
+  IL_002f:  ret
+}}");
+        }
+
+        [Theory]
+        [InlineData(0, 2)]
+        [InlineData(1, 1)]
+        [InlineData(1, 2)]
+        [InlineData(1, 3)]
+        [InlineData(0, 4)]
+        public void CreateSpanIsNotUsedForSpanOfNonSize1Types_NonFullLength(int start, int length)
```

Comment:
Is the test name correct? This test uses a size-1 type.

Reply:
It uses char, which is 2 bytes.